src/controllers/userController.ts

```diff
@@ -0,0 +1,81 @@
+import { prisma } from '@/model/prisma';
+import { Body, errorCode } from '@/utils/utils';
+import { DBError } from '@/validation/custom-error';
+
+export const user = {
+  getById: async (id: number) => {
+    if (isNaN(id)) {
+      throw new DBError('UserId should be a number.', errorCode.INVALID);
```

Comment:
1. Please create middlewares for all validations and validate there, then in a controller you can extract the data and pass it to a service or a repository
2. Wrong params is not a DB error
https://hono.dev/guides/validation